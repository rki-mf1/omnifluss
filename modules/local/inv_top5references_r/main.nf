
process TOP5_REFERENCES {
    tag "$meta.id"

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'docker.io/rkimf1/r-covpipe:89835bd--c37db5b' :
        'docker.io/rkimf1/r-covpipe:89835bd--c37db5b' }"

    input:
    tuple val(meta), path(paf)

    output:
    tuple val(meta), path("*_best_refs.txt"),   emit: txt
    path "versions.yml"                     ,   emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"

    """
    Rscript \\
        ${projectDir}/bin/minimap_stats.r \\
        $paf

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        minimap_stats.r: \$(${projectDir}/bin/minimap_stats.r --version)
    END_VERSIONS
    """

    stub:
    """
    touch top5_best_refs.txt

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        minimap_stats.r: \$(${projectDir}/bin/minimap_stats.r --version)
    END_VERSIONS
    """
}
