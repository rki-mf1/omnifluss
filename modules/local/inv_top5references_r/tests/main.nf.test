// TODO nf-core: Once you have added the required tests, please run the following command to build this file:
// nf-core modules test invtop5referencesr
nextflow_process {

    name "Test Process TOP5_REFERENCES"
    script "../main.nf"
    config "modules/local/inv_top5references_r/tests/nextflow.config"
    process "TOP5_REFERENCES"

    tag "modules"
    tag "modules_local"
    tag "inv_top5references_r"
    tag "TOP5_REFERENCES"
    tag "reference"
    tag "selection"

    test("sarscov2 - paf") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test' ], // meta map
                    [ file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/genome.paf', checkIfExists: true) ]
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    // TODO nf-core: Change the test name preferably indicating the test-data and file-format used but keep the " - stub" suffix.
    test("sarscov2 - paf - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id:'test' ], // meta map
                    [ file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/genome.paf', checkIfExists: true) ]
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
