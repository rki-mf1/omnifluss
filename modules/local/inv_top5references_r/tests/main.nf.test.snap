{
    "sarscov2 - paf": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "genome.paf_best_refs.txt:md5,3a9ea6d336e113a74d7fdca5e7b623fc"
                    ]
                ],
                "1": [
                    "versions.yml:md5,b206184401c7f38576e6037ffa8157df"
                ],
                "txt": [
                    [
                        {
                            "id": "test"
                        },
                        "genome.paf_best_refs.txt:md5,3a9ea6d336e113a74d7fdca5e7b623fc"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,b206184401c7f38576e6037ffa8157df"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.04.1"
        },
        "timestamp": "2024-08-16T12:44:17.007945674"
    },
    "sarscov2 - paf - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "top5_best_refs.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,b206184401c7f38576e6037ffa8157df"
                ],
                "txt": [
                    [
                        {
                            "id": "test"
                        },
                        "top5_best_refs.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,b206184401c7f38576e6037ffa8157df"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.04.1"
        },
        "timestamp": "2024-08-16T12:44:22.789137539"
    }
}